src/App.js

```diff
@@ -2,12 +2,57 @@ import React from 'react';
 
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { GoodsList } from './components/GoogsList/GoodsList';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+class App extends React.Component {
+  state ={
+    goods: [],
+  }
+
+  listOfGoods = (functionName) => {
+    functionName.then(goods => (
+      this.setState({ goods })
+    ));
+  };
+
+  render() {
+    const { goods } = this.state;
+
+    return (
+      <>
+        <h1>Dynamic list of Goods</h1>
+        <button
+          className="button"
+          type="button"
+          onClick={() => {
+            this.listOfGoods(getAll());
+          }}
```

Comment:
pass callback to the function

```suggestion
          onClick={() => this.listOfGoods(getAll)}
```